.github/workflows/pre-commit.yaml

```diff
@@ -0,0 +1,38 @@
+name: pre-commit
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events
+  push:
+    branches: [ "**" ]
+  pull_request:
+    branches: [ "main" ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v3
+
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.8.13
+
+      - name: Install poetry
+        uses: snok/install-poetry@v1
+
+      - name: Install dependencies
+        run: poetry install --no-interaction --no-root
```

Comment:
Consider adding the `--no-dev` option for Poetry. This avoid installing development dependencies that are not necessary to run and deploy the project.

```bash
poetry install --no-interaction --no-root --no-dev
```

Reply:
the pre-commit hooks are coming from the dev requirements. They are not really needed for the application to run in deployment.

Reply:
should I leave this as is? Or what do you suggest?

Reply:
You can safely add the `--no-dev` option. It should not effect the function of pre-commit.

Reply:
done